File: src/main/java/io/github/gravetii/scene/help/WhatIsItScene.java
package io.github.gravetii.scene.help;

import io.github.gravetii.scene.FxDimensions;
import io.github.gravetii.scene.FxScene;

import javafx.geometry.Dimension2D;
import javafx.stage.Stage;

import java.util.Optional;

public class WhatIsItScene extends FxScene {
  private final WhatIsItComponent component;

  public WhatIsItScene(Stage stage) throws Exception {
    super(stage);
    this.component = new WhatIsItComponent();
  }

  @Override
  protected void build() {
    this.showCenter(this.component);
  }

  @Override
  protected String title() {
    return "What is it?";
  }

  @Override
  protected Optional<FxDimensions> preferredDimensions() {
    FxDimensions dimensions = new FxDimensions(new Dimension2D(550, 400));
    return Optional.of(dimensions);
  }
}

File: src/main/kotlin/io/github/gravetii/scene/help/WhatIsItScene.kt
package io.github.gravetii.scene.help

import io.github.gravetii.scene.FxDimensions
import io.github.gravetii.scene.FxScene
import javafx.geometry.Dimension2D
import javafx.stage.Stage

class WhatIsItScene(stage: Stage) : FxScene(stage) {

    private val component = WhatIsItComponent()

    override fun build() = showCenter(component)

    override fun title() = "What is it?"

    override fun preferredDimensions(): FxDimensions = FxDimensions(Dimension2D(575.0, 430.0))

}